python/bindings.cpp

```diff
@@ -539,6 +547,138 @@ PYBIND11_MODULE(_libsonata, m) {
         .def("materialize", &NodeSets::materialize, DOC_NODESETS(materialize))
         .def("update", &NodeSets::update, "other"_a, DOC_NODESETS(update))
         .def("toJSON", &NodeSets::toJSON, DOC_NODESETS(toJSON));
+    
+    py::class_<CompartmentLocation>(m, "CompartmentLocation")
+        .def(py::init<const std::string&>())
+        .def(py::init<const int64_t, const int64_t, const double>(),
+            py::arg("gid"), py::arg("section_idx"), py::arg("offset"))
+        .def_property_readonly("gid", &CompartmentLocation::gid, DOC_COMPARTMENTLOCATION(gid))
+        .def_property_readonly("section_idx",
+                               &CompartmentLocation::sectionIdx,
+                               DOC_COMPARTMENTLOCATION(sectionIdx))
+        .def_property_readonly("offset",
+                               &CompartmentLocation::offset,
+                               DOC_COMPARTMENTLOCATION(offset))
+        .def("toJSON", &CompartmentLocation::toJSON, DOC_COMPARTMENTLOCATION(toJSON))
+        .def("__iter__",
+             [](const CompartmentLocation& self) {
+                 return py::iter(py::make_tuple(self.gid(), self.sectionIdx(), self.offset()));
+             })
+        .def("__eq__", &CompartmentLocation::operator==)
```

Comment:
I'm not convinced all these equality checks are necessary - what is the usecase for it?

Reply:
in case we want to compare them. Also, it comes from the time when you wanted a warning for repeated entries. Since I already wrote some of the code I decided to keep the equalities for CompartmentLocation. Then, for consistency, I added the equality also for the other classes. Overall I do not think it is a lot of additional maintenance